fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsSchedularServiceImpl.java

```diff
@@ -37,42 +40,66 @@
 
     private final SavingsAccountAssembler savingAccountAssembler;
     private final SavingsAccountWritePlatformService savingsAccountWritePlatformService;
-    private final SavingsAccountRepository savingAccountRepository;
     private final SavingsAccountReadPlatformService savingAccountReadPlatformService;
 
     @Autowired
     public SavingsSchedularServiceImpl(final SavingsAccountAssembler savingAccountAssembler,
             final SavingsAccountWritePlatformService savingsAccountWritePlatformService,
-            final SavingsAccountRepository savingAccountRepository,
             final SavingsAccountReadPlatformService savingAccountReadPlatformService) {
         this.savingAccountAssembler = savingAccountAssembler;
         this.savingsAccountWritePlatformService = savingsAccountWritePlatformService;
-        this.savingAccountRepository = savingAccountRepository;
         this.savingAccountReadPlatformService = savingAccountReadPlatformService;
     }
 
-    @CronTarget(jobName = JobName.POST_INTEREST_FOR_SAVINGS)
-    @Override
-    public void postInterestForAccounts() throws JobExecutionException {
-        final List<SavingsAccount> savingsAccounts = this.savingAccountRepository.findSavingAccountByStatus(SavingsAccountStatusType.ACTIVE
-                .getValue());
-        StringBuffer sb = new StringBuffer();
-        for (final SavingsAccount savingsAccount : savingsAccounts) {
-            try {
-                this.savingAccountAssembler.assignSavingAccountHelpers(savingsAccount);
-                this.savingsAccountWritePlatformService.postInterest(savingsAccount);
-            } catch (Exception e) {
-                Throwable realCause = e;
-                if (e.getCause() != null) {
-                    realCause = e.getCause();
-                }
-                sb.append("failed to post interest for Savings with id " + savingsAccount.getId() + " with message "
-                        + realCause.getMessage());
-            }
-        }
-        
-        if (sb.length() > 0) { throw new JobExecutionException(sb.toString()); }
-    }
+	@Transactional
+	@CronTarget(jobName = JobName.POST_INTEREST_FOR_SAVINGS)
+	@Override
+	public void postInterestForAccounts() throws JobExecutionException {
+		final int maxPazeSize = 100;
+		boolean hasErrorOccurred = false;
+		int offSet = 1;
+		Integer initialSize = 100;
+		final Integer status = SavingsAccountStatusType.ACTIVE.getValue();
+		
+		
+		do {
+			Page<SavingsAccountData> savingsAccounts = this.savingAccountReadPlatformService.retrieveSavingsAccount(maxPazeSize, offSet,
```

Comment:
By reading Savings Account Entities page wise, you can avoid lot of db calls.

Reply:
@nazeer1100126 : The reads are page-wise only. I did not understand your concern.

Reply:
Check the complete flow. He is calling postInterestInBatch with Pojos and then he is again reading the entities from DB. My point was to use JPA pagable implementation